src/hooks/useVisibleBalances.ts

```diff
@@ -0,0 +1,21 @@
+import type { SafeBalanceResponse } from '@safe-global/safe-gateway-typescript-sdk'
+import useBalances from './useBalances'
+import useHiddenTokens from './useHiddenTokens'
+
+export const useVisibleBalances = (): {
+  balances: SafeBalanceResponse
+  loading: boolean
+  error?: string
+} => {
+  const balances = useBalances()
+  const hiddenTokens = useHiddenTokens()
+  const { items, fiatTotal } = balances.balances
+
+  return {
+    ...balances,
+    balances: {
+      items: items.filter((item) => !hiddenTokens.includes(item.tokenInfo.address)),
+      fiatTotal,
+    },
```

Comment:
Bear in mind this will filter the list and create a new object on each render. Would be good to memo it.